bitswap/client/internal/session/session.go

```diff
@@ -78,6 +81,41 @@ type ProviderFinder interface {
 	FindProvidersAsync(ctx context.Context, k cid.Cid) <-chan peer.ID
 }
 
+type FindAllProviders struct {
+	Router bsnet.BitSwapNetwork
+}
```

Comment:
If we're going to keep the function here as not taking a `count` the name should get changed from `FindProvidersAsync` to one that doesn't collide with the more widely used function that does take a count so implementations can use both if they want.